index.js

```diff
@@ -37,113 +33,21 @@ function licensee (configuration, path, callback) {
   ) {
     callback(new Error('No licenses or packages allowed.'))
   } else {
-    if (configuration.productionOnly) {
-      // In order to ignore devDependencies, we need to read:
-      //
-      // 1. the dependencies-only dependency graph, from
-      //    `npm ls --json --production`
-      //
-      // 2. the structure of `node_modules` and `package.json`
-      //    files within it, with read-package-tree.
-      //
-      // `npm ls` calls read-package-tree internally, but does
-      // lots of npm-specific post-processing to produce the
-      // dependency tree.  Calling read-package-tree twice, at
-      // the same time, is far from efficient.  But it works,
-      // and doing so helps keep this package small.
-      runParallel({
-        dependencies: readDependencyList,
-        packages: readFilesystemTree
-      }, function (error, trees) {
-        if (error) callback(error)
-        else withTrees(trees.packages, trees.dependencies)
-      })
-    } else {
-      // If we are analyzing _all_ installed dependencies,
-      // and don't care whether they're devDependencies
-      // or not, just read `node_modules`.  We don't need
-      // the dependency graph.
-      readFilesystemTree(function (error, packages) {
-        if (error) callback(error)
-        else {
-          if (configuration.filterPackages) {
-            packages = configuration.filterPackages(packages)
-          }
-          withTrees(packages, false)
+    var arborist = new Arborist({ path })
+    arborist.loadActual({ forceActual: true })
+      .then(function (tree) {
+        var children = Array.from(tree.children.values())
```

Comment:
Note, this is only the packages that happen to be installed in the top-level `node_modules` folder.  (Parent/children = tree = physical folder structure.  Edges = dependencies = logical dependency graph.)

If you intend to get a flat list of _every_ package in the tree, regardless of folder hierarchy or dependency graph (including packages that are unreachable or extraneous), you can instead use `var children = Array.from(tree.inventory.values())`